core/src/runtime/hooks.rs

```diff
@@ -0,0 +1,100 @@
+use core::convert::TryInto;
+use std::collections::HashMap;
+
+use k256::ecdsa::{RecoveryId, Signature, VerifyingKey};
+use k256::elliptic_curve::ops::Invert;
+
+use super::Runtime;
+
+type HookId = u32;
+type BoxedHook<'a> = Box<dyn Fn(HookEnv, &[u8]) -> Vec<Vec<u8>> + 'a>;
+
+// Ensure this value is synced with the values in `zkvm/precompiles/src/io.rs`
```

Comment:
nit: comment is weird, inconsistent use of // and ///

also it should be more like

```
/// The file descriptor through which to access `hook_ecrecover`.
///
/// Note: Ensure this value is synced with the values in `zkvm/precompiles/src/io.rs`.
```